source/common/quic/envoy_quic_client_session.cc

```diff
@@ -102,6 +101,14 @@ EnvoyQuicClientSession::CreateIncomingStream(quic::PendingStream* /*pending*/) {
 
 bool EnvoyQuicClientSession::hasDataToWrite() { return HasDataToWrite(); }
 
+const quic::QuicConnection* EnvoyQuicClientSession::quicConnection() const {
+  return initialized_ ? connection() : nullptr;
```

Comment:
Is it valid to get here when initialized_ is false?  Should we try harder to ensure that Initialize() is called?

Reply:
It is possible when a filter wants to close it upon creation which happens before Initialize(). I added a unit test for that case. If we forget to call Initialize(), QUICHE code will omit tons of errors.